src/python/pants/backend/go/goals/package_binary.py

```diff
@@ -34,13 +35,21 @@ class GoBinaryFieldSet(PackageFieldSet, RunFieldSet):
 
     main: GoBinaryMainPackageField
     output_path: OutputPathField
+    pure: GoPureField
 
 
 @rule(desc="Package Go binary", level=LogLevel.DEBUG)
-async def package_go_binary(field_set: GoBinaryFieldSet) -> BuiltPackage:
+async def package_go_binary(
+    field_set: GoBinaryFieldSet,
+    go_build_context: GoBuildContext,
+) -> BuiltPackage:
+    if field_set.pure.value:
+        go_build_context = dataclasses.replace(go_build_context, cgo_allowed=False)
```

Comment:
It would be nice if the modified context could be use for all subsequent `Get`'s without having to use the dict-style syntax for each and every one. Context manager syntax might be useful:

```
with engine.set_param({go_build_context: GoBuildContext}):
   ...
```